api/proto/config/devmodel.proto

```diff
@@ -56,3 +86,57 @@ message SystemAdapter {
 	// and used for app instances
 	string logicalName = 6;
 }
+
+enum PhyIoType {
+        PhyIoNoop = 0;
+        PhyIoEth = 1;
+        PhyIoUSB = 2;
+        PhyIoCOM = 3;
+        PhyIoAudio = 4;
+        PhyIoWLAN = 5;
+        PhyIoWWAN = 6;
+        PhyIoHDMI = 7;
+        PhyIoOther = 255;
+}
+
+// How does EVE should use them, for what purpose it is for
+enum PhyIoMemberUsage {
+	PhyIoUsageNone      = 0;
+	PhyIoUsageMgmt      = 1;
+	PhyIoUsageShared    = 2;
+	PhyIoUsageDedicated = 3;
+}
+
+// PhysicalIO:
+//    Absolute low level description of physical buses and ports that are
+//    available on given platfrom.
+//    Collection of these IOs, connstitue what we would call as hardware
+//    model. Each physical IO is manageable and visible to EVE software, and
+//    it can be further configured to either provide IP connectivity or
+//    directly be given to workloads
+message PhysicalIO {
+	PhyIoType ptype           = 1;
+
+	// physical label typically printed on box.
+	// Example Eth0, Eth1, Wifi0, ComA, ComB
+	string phylabel           = 2;
+
+	// the hardware bus address
```

Comment:
What's the keys in the map?

Reply:
Fine with me, but then we should define the set which are networking (because networking can be used for systemAdapters and today we have specific checks for this in Eve by comparing with "CioEth". If such checks aren't needed on the cloud and UI, then we can keep the definition of the set on the Eve side. But I suspect the UI also check which things are ethernets/networks. Makes sense to check that with Sahil. Or we can defer adding the enums until after we check with him

Reply:
I added your comment from other thread here. I agree we need other, we can add wrapper function in utils, which provide that checks are returns it is networking interfaces if it is either Eth/Wifi/Wlan etc.

I don't know how to do that in protobuf yet.

Reply:
Need some more comments so I can understand how to use this.